### PR TITLE
Fixed errcheck issues in server/channels/api4/license_local.go

### PR DESCRIPTION
**Pull request by Aryakoste**
#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/28449
Jira https://mattermost.atlassian.net/browse/MM-60793
#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: server/channels/api4/license_local.go [hunk 1]
```diff
@@ -54,7 +54,9 @@ func localAddLicense(c *Context, w http.ResponseWriter, r *http.Request) {
 	defer file.Close()
 
 	buf := bytes.NewBuffer(nil)
-	io.Copy(buf, file)
+	if _, err := io.Copy(buf, file); err != nil {
+		c.Logger.Warn("Failed to copy file content to buffer", mlog.Err(err))
```
Comment (hanzei) [inline server/channels/api4/license_local.go:58]: Failing to copy the input puffer seems like a blocking error to me. I think we should return an error to the client. Is there an existing error ID that we can reuse?

**Review by hanzei — CHANGES_REQUESTED** [review]
Thanks for the PR :+1:

**Comment by hanzei** [comment]
@Aryakoste Do you have any questions about the feedback?

**Comment by Aryakoste** [comment]
@hanzei Changes done.

**Review by hanzei — APPROVED** [review]
LGTM :+1:

**Review by calebroseland — APPROVED** [review]

**Pull request merged**